### PR TITLE
[Merged by Bors] - feat: `a * a ^ (n - 1) = a ^ n`

### PR DESCRIPTION
**Pull request by YaelDillies**
From LeanAPAP

---

- [x] depends on: #9551

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by riccardobrasca** [comment]
I guess this needs a merge master.

File: Mathlib/Algebra/Group/Defs.lean [hunk 1]
```diff
@@ -669,7 +669,7 @@ attribute [to_additive existing toNatSMul] Monoid.toNatPow
 
 section
 
-variable {M : Type*} [Monoid M]
+variable {M : Type*} [Monoid M] {n : ℕ}
```
Comment (riccardobrasca) [inline Mathlib/Algebra/Group/Defs.lean:672]: This line is probably not needed.

**Review by riccardobrasca — APPROVED** [review]
Thanks!

bors d+

File: Mathlib/Algebra/Group/Defs.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Algebra/Group/Defs.lean:672]:
```suggestion
variable {M : Type*} [Monoid M]
```

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**